server/common/Configuration.java

```diff
@@ -61,25 +63,25 @@
 
 public class Configuration {
 
-    private final Path dataDir;
-    private final int port;
+    private final Server server;
+    private final Storage storage;
```

Comment:
we support new sub-section `Server` and `Storage` for network related configurations and storage-layer options